pkg/controller/datadogagent/utils.go

```diff
@@ -159,6 +159,28 @@ func getAgentContainer(dda *datadoghqv1alpha1.DatadogAgent) (*corev1.Container,
 	if err != nil {
 		return nil, err
 	}
+
+	udpPort := corev1.ContainerPort{
+		ContainerPort: datadoghqv1alpha1.DefaultDogstatsdPort,
+		Name:          "dogstatsdport",
+		Protocol:      corev1.ProtocolUDP,
+	}
+
+	if agentSpec.Config.HostPort != nil {
+		// Create the host port configuration
+		udpPort.HostPort = *agentSpec.Config.HostPort
+		// If HostNetwork is enabled, set the container port
+		// and the DD_DOGSTATSD_PORT environment variable to match the host port
+		// so that Dogstatsd can be reached on the port configured in HostPort
```

Comment:
@ahmed-mez Is this descriptive enough? Or maybe a bit too much? Let me know if you want any changes there :)

Reply:
it looks good, thanks 💯